packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -490,11 +490,40 @@ class SelectableRegionState extends State<SelectableRegion> with TextSelectionDe
   }
 
   void _handleRightClickDown(TapDownDetails details) {
+    final Offset? previousSecondaryTapDownPosition = lastSecondaryTapDownPosition;
+    final bool toolbarIsVisible = _selectionOverlay?.toolbarIsVisible ?? false;
     lastSecondaryTapDownPosition = details.globalPosition;
     widget.focusNode.requestFocus();
-    _selectWordAt(offset: details.globalPosition);
-    _showHandles();
-    _showToolbar(location: details.globalPosition);
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.windows:
+        _selectStartTo(offset: details.globalPosition);
+        _selectEndTo(offset: details.globalPosition);
+        _showHandles();
+        _showToolbar(location: details.globalPosition);
+      case TargetPlatform.iOS:
+        _selectWordAt(offset: details.globalPosition);
+        _showHandles();
+        _showToolbar(location: details.globalPosition);
+      case TargetPlatform.macOS:
+        if (previousSecondaryTapDownPosition == lastSecondaryTapDownPosition && toolbarIsVisible) {
```

Comment:
This seems weird. of user click a slightly different place but still in the same word. will they be able to toggle the toolbar?

Reply:
In that case the toolbar should not be able to be toggled. Only if it is in the same place will it be toggled. If it is anywhere else the toolbar is re-shown at the new position. I tested this in the native notes app, as well as on the web.

Reply:
sounds good then